### PR TITLE
Upgrade from Chromium 86 to Chromium 87 (1.17.x).

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves brave/brave-browser#11534
Related PR: brave/brave-browser#12564

Uplifts from:
#6554
#6986
#6989
#6998
#6997
#7002
#7037
#7052
#7055
#7058
#7061
#7054

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by mkarolin** [comment]
@darkdh Tor migration has not been uplifted to 1.17.x, so I had to make an additional change for cr87 here. Could you, please, take a look at 4e04b475c48a0df0b2bf13a46da2748421ca57a4 and see if it looks ok? Thanks.

**Comment by kjozwiak** [comment]
Going to restart `Win x64` as it seems like we had some issues with the instance disconnecting.

File: components/services/tor/public/interfaces/BUILD.gn
Comment (darkdh) [inline components/services/tor/public/interfaces/BUILD.gn:21]: we also need `//brave/common/tor` in  `traits_public_deps`

Comment (darkdh) [reply components/services/tor/public/interfaces/BUILD.gn:21]: see upstream changes of `proxy_mojom_traits` in https://chromium.googlesource.com/chromium/src/+/4b3aa3c059810d00033e43de06d5fe856da7ab3d%5E%21/

Comment (mkarolin) [reply components/services/tor/public/interfaces/BUILD.gn:21]: @darkdh, if I do that I get
<pre>
ERROR Dependency cycle:
  //brave/components/services/tor/public/interfaces:interfaces ->
  //brave/common/tor:tor ->
  //brave/components/services/tor/public/interfaces:interfaces
</pre>

Comment (darkdh) [reply components/services/tor/public/interfaces/BUILD.gn:21]: we should remove `//brave/components/services/tor/public/interfaces:interfaces` in "//brave/common/tor". That dependency seems unnecessary

Comment (mkarolin) [reply components/services/tor/public/interfaces/BUILD.gn:22]: Sounds good. Fixed in 657403db7ae13f4f1d02df3cf34390ae33584266

**Review by darkdh — APPROVED** [review]

**Comment by kjozwiak** [comment]
Looks like `test-install` failed on `Win x64` as per https://ci.brave.com/job/pr-brave-browser-cr87-1.17.x-windows/4/execution/node/272/log/. However, this is a known intermittent issue and doesn't block uplift.

@mkarolin @darkdh thanks for getting the above done so quickly re: missing deps for Tor. Going to start the uplift process so we can start getting builds with C87 on `1.17.x` 👍

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.17.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
